CoroutineTemplate/domain/src/main/java/co/nimblehq/coroutine/domain/storage/BaseSharedPreferences.kt

```diff
@@ -0,0 +1,40 @@
+package co.nimblehq.coroutine.domain.storage
+
+import android.content.SharedPreferences
+import co.nimblehq.coroutine.domain.extension.execute
+
+abstract class BaseSharedPreferences {
+
+    protected lateinit var sharedPreferences: SharedPreferences
+
+    protected fun <T> set(key: String, value: T?) {
+        sharedPreferences.execute {
+            when (value) {
+                null -> it.remove(key)
```

Comment:
I'm curious will we use `SharedPreferences.set(key, null)` in case we want to remove that key? @minhnimble

Reply:
yeah that is the idea @lydiasama. Or is there any better way to do so? 🤔

Reply:
Can we create another function to do that purpose especially? 
e.g. `fun remove(key: String)`

Reply:
@lydiasama updated 🙏 you can check again now 😁